src/world.rs

```diff
@@ -1,100 +1,67 @@
-extern crate rand;
-
-use std::collections::HashMap;
+use itertools::Itertools;
+use std::collections::{HashMap, HashSet};
 use std::fmt;
+use rand::random;
 
 type Point = (isize, isize);
 
-
-#[derive(Debug, PartialEq, Clone)]
-enum CellState{
-    Alive,
-    Dead
-}
-
-
-impl rand::Rand for CellState {
-    #[inline]
-    fn rand<R:rand::Rng>(_: &mut R) -> CellState {
-        match rand::random() {
-            true => CellState::Alive,
-            false => CellState::Dead
-        }
-    }
-}
-
-
 #[derive(Debug, Clone)]
 pub struct World {
     width: isize,
     height: isize,
-    cells: HashMap<Point, CellState>
+    generation: HashSet<Point>,
+    neighbours: HashMap<Point, usize>,
+    age: usize,
 }
 
-
 impl World {
     pub fn new(width: isize, height: isize)  -> World {
-        let mut world = World{width: width, height: height, cells: HashMap::new()};
-        for y in 0..height { for x in 0..width {
-            world.cells.insert((x, y), CellState::Dead);
-        }}
-        world
+        World { width: width, height: height, age: 0, generation: HashSet::new(),
+                neighbours: HashMap::with_capacity((width * height * 8) as usize) }
     }
 
-    pub fn evolve(&self) -> World {
-        let mut new_world = self.clone();
-        for y in 0..self.height { for x in 0..self.width {
-            new_world.set_state(x, y, self.decide_fate(x, y))
-        }}
-        new_world
-
+    pub fn seed(mut self) -> World {
+        self.generation = (0..self.width).cartesian_product(0..self.height)
```

Comment:
Yeah toolbelt guide is necessary. But so far you can subscribe on http://this-week-in-rust.org/ and get updates about rust related stuff almost every week. http://blog.servo.org/ also pretty exciting reading. And off course http://www.reddit.com/r/rust/ . Code convention, style guide, whatever you can find at https://github.com/rust-lang/rust-guidelines . Have fun :smile_cat: